File: backend/src/main/java/com/dev/backend/service/ProductService.java
package com.dev.backend.service;

import java.util.Date;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.dev.backend.enitity.Product;
import com.dev.backend.repository.ProductRepository;

@Service
public class ProductService {
    
    @Autowired
    private ProductRepository productRepository;

    public List<Product> searchAll(){
        return productRepository.findAll();
    }

    public Product insert(Product product){
        product.setDateCreation(new Date());
        Product productNew = productRepository.saveAndFlush(product);
        return productNew;
    }

    public Product alter(Product product){
        product.setDateUpdate(new Date());
        return productRepository.saveAndFlush(product);
    }

    public void delete(Long id){
        Product product = productRepository.findById(id).get();
        productRepository.delete(product);
    }
}

File: frontend/src/service/register/ProductService.js
import axios from 'axios';

export class ProductService{

    url = process.env.REACT_APP_URL_API;

    product(){
        return axios.get(this.url+'/product/');
    }

    insert(product){
        return axios.post(this.url+'/product/', product);
    }

    alter(product){
        return axios.put(this.url+'/product/', product);
    }

    delete(id){
        return axios.delete(this.url+'/product/'+id);
    }
}